x-pack/plugins/security/server/authentication/providers/kerberos.ts

```diff
@@ -332,11 +335,14 @@ export class KerberosAuthenticationProvider extends BaseAuthenticationProvider {
    * Extracts `Negotiate` challenge from the list of challenges returned with Elasticsearch error if any.
    * @param error Error to extract challenges from.
    */
-  private getNegotiateChallenge(error: LegacyElasticsearchError) {
+  private getNegotiateChallenge(error: errors.ResponseError) {
+    // We extract headers from the original Elasticsearch error and not from the top-level `headers`
```

Comment:
__note:__ well, in fact it's NodeJS that merges these headers, but we can pretend we don't know that :slightly_smiling_face: